anvil_consortium_manager/adapter.py

```diff
@@ -0,0 +1,40 @@
+"""Contains default adapter for workspaces."""
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.module_loading import import_string
+
+from . import tables
+
+
+class DefaultWorkspaceAdapter(object):
```

Comment:
I'm not sure what you mean by "true base model" so agree, we can discuss tomorrow.

Reply:
I implemented an abstract base model (`AbstractWorkspaceData`) and then a default model (`DefaultWorkspaceData`) that inherits `AbstractWorkspaceData`. I wanted the default model so that the app could be used without implementing your own custom workspace data model, if you didn't have any extra information to store. This has the side effect of creating an extra table (for `DefaultWorkspaceData`) and an extra field for a workspace object (`workspace.defaultworkspacedata`), which exists even if you are defining your own custom model. There is also an empty form with no fields (to which `workspace` could be added if we start requiring the forms to have a `workspace` field). Does this sound reasonable, or do you see any big red flags with the `DefaultWorkspaceData` model?

Reply:
I was actually referring to an abstract base model for the adapter, not the workspace data model. 
If adding a default workspace data model like you have here makes things simpler I don't see any problem with it.

Reply:
So something like `AbstractBaseAdapter` with everything set to `None` and all the methods defined, and then also have the `DefaultWorkspaceAdapter` model and just set the fields to the defaults?

Reply:
I ended up keeping the `DefaultWorkspaceData` model. That way, the logic in the views is a little easier, since you don't have to check if something is set in the adapter and then if it is, do something with it.

Reply:
Also implemented the abstract base class for the adapter.